server/src/services/customers.ts

```diff
@@ -45,4 +49,38 @@ const addCustomer = async (
     ...customer,
   });
 
-export default {getCustomer, getCustomerWithGpayId, addCustomer};
+/**
+ * Updates a customer with the specified customerId by modifying the fields in fieldsToUpdate.
+ * Throws an error if fieldsToUpdate contains fields that are not allowed to be modified,
+ * or if the customer does not exist.
+ * @param customerId Id of the customer to update
+ * @param fieldsToUpdate Fields of the customer to be updated
+ */
+const updateCustomer = async (
+  customerId: number,
+  fieldsToUpdate: Partial<CustomerPayload>
+) => {
+  // Check that customer exists
+  await customerStorage.getCustomer(customerId);
```

Comment:
Is this query necessary? Won't there already be an error thrown during update if a customer with given customerId doesn't exist?

Reply:
@karenfrilya97 
Actually right now datastore `updateEntity` does not throw an appropriate error if customer doesn't exist.
Indeed some kind of error will be thrown when updating a customer with invalid customerId because in `updateEntity` we have:
 ```
  const key = datastore.key([kind, id]);
  const [data] = await datastore.get(key);
```
And `data` will be `undefined` if the `id` in the key does not exist. So right now our `updateEntity` will throw errors because then it would be working with something that's undefined.
if you think it's better to not do the check in services layer then I guess I can modify `updateEntity` to throw an appropriate error when `data === undefined`!

Reply:
@hzxcaryn Personally, I prefer to throw the appropriate error in `updateEntity` function. 

I think that updating something that doesn't exist should not be allowed. This would also reduce repetition in the services layer.

Also, even though get function is relatively fast and has O(1) complexity, I prefer not to make an extra trip to the database unless it's important.

Reply:
@karenfrilya97 Oh I already changed `updateEntity` to call `getEntity` when I resolved the comment a few days ago, forgot to mention 😓 This means that `updateEntity` would throw the same error as `getIdentity` if entity does not exist.